diff --git a/src/containers/extension-library.jsx b/src/containers/extension-library.jsx
index b46d992c0..47277ff9f 100644
--- a/src/containers/extension-library.jsx
+++ b/src/containers/extension-library.jsx
@@ -37,7 +37,7 @@ class ExtensionLibrary extends React.PureComponent {
             url = prompt(this.props.intl.formatMessage(messages.extensionUrl));
         }
         if (id && !item.disabled) {
-            if (this.props.vm.extensionManager.isExtensionLoaded(url)) {
+            if (this.props.vm.extensionManager.isExtensionLoaded(id)) {
                 this.props.onCategorySelected(id);
             } else {
                 this.props.vm.extensionManager.loadExtensionURL(url).then(() => {
@@ -51,6 +51,14 @@ class ExtensionLibrary extends React.PureComponent {
             rawURL: extension.iconURL || extensionIcon,
             ...extension
         }));
+        extensionLibraryContent.forEach(extension => {
+            if (extension.translationMap) {
+                Object.assign(
+                    this.props.intl.messages,
+                    extension.translationMap[this.props.intl.locale]
+                );
+            }
+        });
         return (
             <LibraryComponent
                 data={extensionLibraryThumbnailData}
