diff --git a/webpack.config.js b/webpack.config.js
index 00ad0983f..2f07b4de8 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -22,6 +22,12 @@ const base = {
         host: '0.0.0.0',
         port: process.env.PORT || 8601
     },
+    watchOptions: {
+        ignored: [
+            /node_modules([\\]+|\/)+(?!scratch-vm)/,
+            /\scratch-vm([\\]+|\/)node_modules/
+        ]
+    },
     output: {
         library: 'GUI',
         filename: '[name].js',
