diff --git a/src/components/stage-header/stage-header.jsx b/src/components/stage-header/stage-header.jsx
index 4084f6bfc..de6f65785 100644
--- a/src/components/stage-header/stage-header.jsx
+++ b/src/components/stage-header/stage-header.jsx
@@ -68,7 +68,7 @@ const StageHeaderComponent = function (props) {
         const stageButton = showBranding ? (
             <div className={styles.embedScratchLogo}>
                 <a
-                    href="https://scratch.mit.edu"
+                    href="https://lab.yengawa.com/project/scratch-microbit-more/"
                     rel="noopener noreferrer"
                     target="_blank"
                 >
diff --git a/src/playground/render-gui.jsx b/src/playground/render-gui.jsx
index 0f15fbfa7..993894bba 100644
--- a/src/playground/render-gui.jsx
+++ b/src/playground/render-gui.jsx
@@ -8,7 +8,7 @@ import HashParserHOC from '../lib/hash-parser-hoc.jsx';
 import log from '../lib/log.js';
 
 const onClickLogo = () => {
-    window.location = 'https://scratch.mit.edu';
+    window.location = 'https://lab.yengawa.com/project/scratch-microbit-more/';
 };
 
 const handleTelemetryModalCancel = () => {
